modules/ebi-metagenomics/keggpathwayscompleteness/main.nf

```diff
@@ -0,0 +1,51 @@
+
+process KEGGPATHWAYSCOMPLETENESS {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/kegg-pathways-completeness:1.0.5--pyhdfd78af_0':
+        'biocontainers/kegg-pathways-completeness:1.0.5--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(filtered_tbl), path(ko_list)
+
+    output:
+    tuple val(meta), path("*_contigs.tsv") , emit: kegg_contigs
+    tuple val(meta), path("*_pathways.tsv"), emit: kegg_pathways
+    path "versions.yml"                    , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def kegg_input = ko_list ? "-l ${ko_list}" : "-i ${filtered_tbl}"
```

Comment:
I'm wondering is it possible to add a WARN or check if both files are presented as input..

Reply:
I added a check:
```
if (ko_list && filtered_tbl){
        log.warn("Both \$ko_list and \$filtered_tbl were given as input types, will fall back to using \$filtered_tbl i.e. ${filtered_tbl}");
    }
```

Which prints this in the nextflow.log if it's true:
`Jul-09 10:20:56.720 [Actor Thread 2] WARN  nextflow.Nextflow - Both $ko_list and $filtered_tbl were given as input types, will fall back to using $filtered_tbl i.e. test_filtered.tbl`

Reply:
but you will use `ko_list` if both are presented, don't you?